nwb_conversion_tools/datainterfaces/ecephys/axona/axonadatainterface.py

```diff
@@ -296,3 +425,211 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+    footer_size = len("\r\ndata_end\r\n")
+    header_size = len(get_header_bstring(filename))
+    num_bytes = os.path.getsize(filename) - header_size - footer_size
+
+    # .eeg files are int8, .egf files are int16
+    if str(filename).split(".")[1][0:3] == "egf":
+        lfp_dtype = ">i2"
+        num_bytes = num_bytes // 2
+
+    eeg_data = np.memmap(
+        filename=filename,
+        dtype=lfp_dtype,
+        mode="r",
+        offset=len(get_header_bstring(filename)),
+        shape=(1, num_bytes),
+    )
+
+    return eeg_data
+
+
+def get_all_filenames(filename):
+    """
+    Read LFP filenames of `.eeg` or `.egf` files in filename's directory.
+    E.g. if filename='/my/directory/my_file.eeg', all .eeg channels will be
+    appended to the output.
+
+    Parameters:
+    -----------
+    filename : path-like
+        Full filename of either .egg or .egf file
+
+    Returns:
+    --------
+    path_list : list
+        List of filenames
+    """
+
+    suffix = Path(filename).suffix[0:4]
+    current_path = Path(filename).parent
+
+    path_list = [cur_path.name for cur_path in Path(filename).parent.rglob("*" + suffix + "*")]
+
+    return path_list
+
+
+def read_all_eeg_file_lfp_data(filename):
+    """
+    Read LFP data from all Axona `.eeg` or `.egf` files in filename's directory.
+    E.g. if filename='/my/directory/my_file.eeg', all .eeg channels will be conactenated
+    to a single np.array (chans x nobs). For .egf files substitude the file suffix.
+
+    Parameters:
+    -------
+    filename (Path or Str):
+        Full filename of Axona `.eeg` or `.egf` file.
+
+    Returns:
+    -------
+    np.array (chans x obs)
+    """
+
+    filename_list = get_all_filenames(filename)
+    parent_path = Path(filename).parent
+
+    eeg_memmaps = list()
+    sampling_rates = set()
+    for fname in filename_list:
+
+        sampling_rates.add(get_eeg_sampling_frequency(parent_path / fname))
+
+        eeg_memmaps.append(read_eeg_file_lfp_data(parent_path / fname))
+
+    assert len(sampling_rates) < 2, "File headers specify different sampling rates. Cannot combine EEG data."
+
+    eeg_data = np.concatenate(eeg_memmaps, axis=0)
+
+    return eeg_data
+
+
+def AxonaLFPNumpyExtractorWrapper(filename):
+    """
+    Wrapper for instantiating a NumpyRecordingExtractor given an `.eeg` or `.egf` filename.
+    """
+    return se.NumpyRecordingExtractor(
+        timeseries=read_all_eeg_file_lfp_data(filename), sampling_frequency=get_eeg_sampling_frequency(filename)
+    )
+
+
+class AxonaLFPDataInterface(AxonaRecordingExtractorInterface):
+    """..."""
+
+    @classmethod
+    def get_source_schema(cls):
+        return dict(
+            required=["filename"],
+            properties=dict(filename=dict(type="string")),
+            type="object",
+            additionalProperties=False,
+        )
+
+    def __init__(self, **source_data):
+        self.recording_extractor = AxonaLFPNumpyExtractorWrapper(**source_data)
```

Comment:
I think it's better to explicitly state what the inputs are rather than just passing kwargs

Reply:
I agree.